src/ImageProcessor.Web.Plugins.PostProcessor/PostProcessorApplicationEvents.cs

```diff
@@ -55,7 +55,7 @@ public static void SetPostProcessingTimeout(int milliseconds)
         /// </param>
         private static void PostProcess(object sender, PostProcessingEventArgs e)
         {
-            e.ImageStream = PostProcessor.PostProcessImage(e.Context, e.ImageStream, e.ImageExtension);
+            e.ImageStream = PostProcessor.PostProcessImageAsync(e.Context, e.ImageStream, e.ImageExtension).Result;
```

Comment:
I just had to read the documentation and saw the `OnPostProcessing` event is awaitable (http://imageprocessor.org/imageprocessor-web/imageprocessingmodule/#events), so should we change this to an async method and await the `PostProcessImageAsync` call?

Reply:
If we can yeah, that would be better.

Reply:
Just tested it and we can´t: mixing events and async just won't work together... In this case the `ImageStream` is already disposed, plus the event delegate is run in a fire-and-forget manner (so the image is already cached when the `PostProcessImageAsync` returns).